### ISSUE TITLE
99th_percentile_frame_rasterizer_time_millis in [flutter_gallery_ios__transition_perf, flutter_gallery__transition_perf_e2e_ios32] regression

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#68316]
https://flutter-flutter-perf.skia.org/e/?begin=1602654448&end=1602714273&keys=X1ff0283333c366cd3e7c124d9966ab3e&num_commits=50&request_type=1&xbaroffset=22001

Started at commit https://github.com/flutter/flutter/pull/68128, but I'm wondering if it could be due to https://github.com/flutter/flutter/pull/68148.

**Comment by gaaclarke** [comment]
@jonahwilliams Is there an expected performance change from #68128?

@kf6gpe In this issue, too, you suspect the engine roll.  Do you think there is an error in our performance tests where attribution is incorrectly recorded?

**Comment by jonahwilliams** [comment]
The change linked only uninstalls applications after the driver script has run. It should not be affecting performance metrics

**Comment by gaaclarke** [comment]
@jonahwilliams Unless the time it takes to uninstall the app is now being incorporated into the measurement.  The only other explanation is that our performance tests are incorrect, right?

**Comment by jonahwilliams** [comment]
The apps were being uninstalled before, so that seems unlikely. There was only a period of 5-10 commits where they weren't always uninstalled

**Comment by gaaclarke** [comment]
This appears to be another case of misattribution.  The commit that perf tool points to doesn't seem to create a problem when tested locally.  Here were the results I got:

33903c17e18f6ebbc523f472ebdc02cdf0356061 "99th_percentile_frame_rasterizer_time_millis": 47.611,
767fb98a7b3b75bb6545cd707b9ad5103ae12df7 "99th_percentile_frame_rasterizer_time_millis": 47.24

**Comment by gaaclarke** [comment]
I tested https://github.com/flutter/flutter/pull/68148 locally and saw a change of 5ms which doesn't match the 30ms increase shown in skia performance.  I'll dig deeper.

ed2a4809dae2c8faa643a55897d824c2006627fd - "99th_percentile_frame_rasterizer_time_millis": 46.346,
12bea61c2902784fec73b7f5f2ff3415f873fead - "99th_percentile_frame_rasterizer_time_millis": 41.092,

**Comment by gaaclarke** [comment]
Hey @keyonghan I'm getting a 30ms jump in skia performance monitoring that I can't reproduce locally.  Is there something messing with the devicelab device?  Amber alert, etc?

**Comment by keyonghan** [comment]
Unfortunately we currently have no way to remotely monitor/dismiss popped up windows for the ios32 device..

**Comment by jonahwilliams** [comment]
What device are you running on locally @gaaclarke ?

**Comment by gaaclarke** [comment]
That's not what the graph is showing, the graph is showing that there is a regression for the 32bit device and 64bit device.  Which makes it sound less likely to be a hardware issue, but still doesn't explain why I can't reproduce it locally.

**Comment by gaaclarke** [comment]
Here's a recap of where we are:
1) The author of the commit expressing the regression is confident that it isn't to blame.  I agree it is very very unlikely the cause.
1) Local testing of that commit doesn't produce the regression
1) An engine roll was suspected, local testing shows no problem there
1) The regression appears on 32bit devices and 64bit devices which seems to suggest it isn't a hardware problem with one device (things like amber alerts would show up on both devices, however).  It also lessons the likelihood that the problem is device specific.

There is a chance that this problem for some reason don't show up in my iOS 13 iPhone 8 but does show up in both the devicelab devices.  @keyonghan what is the 64bit device we are using in devicelab?

**Comment by keyonghan** [comment]
All other iphones are with 64bit ([go/flutter-devicelab-hardware](go/flutter-devicelab-hardware)).

**Comment by gaaclarke** [comment]
I'm not sure how to proceed here.  The commit associated with the regression is one that the author doesn't believe caused a regression; I agree with the assessment; local testing doesn't show a regression, in fact it shows an improvement; yet the regression shows up on 2 of our devicelab units.  I have no plausible explanation why the commit in question would potentially cause a regression let alone cause a regression on one type of device and not another.

I don't even know why the engine roll was suspected since it has no proximity to the appearance of the regression.  If someone could suggest a plausible hypothesis for this regression I could continue exploring this otherwise I'm just stabbing around in the dark.

**Comment by gaaclarke** [comment]
I'm going to revert https://github.com/flutter/flutter/pull/68128 and see if the number drop back down.

**Comment by jonahwilliams** [comment]
Please do not revert devicelab stability fixes

**Comment by jonahwilliams** [comment]
If the tests cannot function properly with the IPA/APK being uninstalled after each run, they need to be disabled.

**Comment by cbracken** [comment]
The benchmarks clearly show a regression at that commit. Looking at it, I agree that if that commit is responsible then there's something wrong with the benchmark itself. We should revert the commit, let it sit for five or six commits to check the efffect, then re-land this commit.

If, during the revert period, we see the benchmarks revert back to where they were, we should investigate what's wrong with the benchmark itself.

**Comment by gaaclarke** [comment]
You know, now that I think about it.  If a regression in performance happened after we started uninstalling the app, one likely culprit would be deleting the shader cache which could result in slower performance.  Thus a dip in performance would be expected.

**Comment by gaaclarke** [comment]
I was finally able to reproduce this at 33903c17e18f6ebbc523f472ebdc02cdf0356061!  it requires running the test multiple times,  I went from ~46ms to ~6ms because of the shader cache.

So, this test needs the be re-baselined and we need to add a new test that measures performance with a full shader cache.  I'll file a bug for that and close the one I created about misattribution.  I'll also close the PR of the revert, thanks @jonahwilliams and @cbracken  for the help.  That one was a doozie.
